frontend/src/app/shared/logged-on.guard.ts

```diff
@@ -1,20 +1,52 @@
 import { inject } from "@angular/core";
 import { ActivatedRouteSnapshot, CanActivateFn, Router } from "@angular/router";
 import { AuthService } from "@services/auth.service";
+import { ToastService } from "@services/toast.service";
 import { filter, map } from "rxjs";
 
 export const LoggedOnGuard: CanActivateFn = (route: ActivatedRouteSnapshot) => {
     const authService = inject(AuthService);
+    const toastService = inject(ToastService);
     const router = inject(Router);
 
     return authService.currentUser$.pipe(
         filter((user) => user !== undefined),
         map((user) => {
+            toastService.show({
```

Comment:
You need to move `if (user === null) {` or some other check here. Currently, the toast is shown whenever this guard is triggered.